### PR TITLE
updates for cloud solve

### PR DESCRIPTION
**Pull request by Affie**
- buildTreeFromOrdering: dispatch differently on InMemoryTypes

EDIT: These parts were already merged
- getEliminationOrder: uses getAdjacencyMatrixSparse
- InMemDFGType set to DFG.GraphsDFG{SolverParams}

### CODE REVIEW

**Comment by dehann** [comment]
hang on with this PR till @GearsAD is happy...

**Comment by dehann** [comment]
just want to make sure @GearsAD does not forget about this PR

**Comment by dehann** [comment]
should we target this for IIF v0.7.x or IIF v0.8.x?

**Comment by Affie** [comment]
Needed DFG 0.4.1

**Comment by Affie** [comment]
I don't know how you feel about adding Neo4j to IIF for tests. I think this is where the test repo comes in?
With this you should be able to run a solve on a CloudGraphsDFG directly for end to end test.
Although, it should currently run (although I think there are serialization issues with some types), I suspect it will fail numerically because of bugs in the driver.
This will now work
```julia
fg = initfg(CloudGraphsDFG{SolverParams}(...))
...
tree, smt, hist = solveTree!(fg)
```

**Comment by Affie** [comment]
Ok, reason I asked is patch coverage check that fails can only be fixed with CGDFG test.

**Comment by dehann** [comment]
> patch coverage check that fails can only be fixed with CGDFG test

ah cool, although i think actually it fails for decreases and passes for increases in test coverage.  Or at least that is how i interpreted it before.

File: src/JunctionTree.jl [hunk 1]
```diff
@@ -286,6 +286,45 @@ function buildTreeFromOrdering!(dfg::G,
   return tree
 end
 
+
+"""
+    $SIGNATURES
+
+Build Bayes/Junction/Elimination tree from a given variable ordering.
+"""
+function buildTreeFromOrdering!(dfg::DFG.AbstractDFG,
```
[the remark this answers is not in the document]
Comment (dehann) [reply src/JunctionTree.jl]: was inmemory vs cloud

**Pull request merged**